### PR TITLE
Tschie/remove controlsfx

### PR DESCRIPTION
**Pull request by tschie**
Prepares the samples to support java 9.

- Update gradle/api versions (4.2+ needed for Java 9)
- Update gradle wrapper
- Removed controlsfx dependency (doesn't work for Java 9)

### CODE REVIEW

**Comment by tschie** [comment]
@john0005 Can you please review?

File: build.gradle [hunk 1]
```diff
@@ -14,7 +14,7 @@ buildscript {
     }
 }
 
-arcgis.version = '100.2.0-1877'
+arcgis.version = '100.2.0-1887'
```
Comment (john0005) [inline build.gradle]: Latest version is 1886, as of today.

File: src/main/java/com/esri/samples/analysis/viewshed_location/ViewshedLocationController.java [hunk 2]
```diff
@@ -76,7 +76,7 @@ public void initialize() {
     sceneView.getAnalysisOverlays().add(analysisOverlay);
 
     // create a listener to update the viewshed location when the mouse moves
-    EventHandler<MouseEvent> mouseMoveEventHandler = new EventHandler<MouseEvent>() {
+    EventHandler<MouseEvent> mouseMoveEventHandler = new EventHandler<>() {
```
Comment (john0005) [inline src/main/java/com/esri/samples/analysis/viewshed_location/ViewshedLocationController.java:79]: Could setup event to be like so,
```java
EventHandler<MouseEvent> mouseMoveEventHandler = event -> {
      Point2D point2D = new Point2D(event.getX(), event.getY());
      ListenableFuture<Point> pointFuture = sceneView.screenToLocationAsync(point2D);
      pointFuture.addDoneListener(() -> {
        try {
          Point point = pointFuture.get();
          PointBuilder pointBuilder = new PointBuilder(point);
          pointBuilder.setZ(point.getZ() + 50.0);
          viewshed.setLocation(pointBuilder.toGeometry());
          // add listener back
          sceneView.setOnMouseMoved((EventHandler<MouseEvent>) event.getSource());
        } catch (InterruptedException | ExecutionException e) {
          e.printStackTrace();
        }
      });
      // disable listener until location is updated (for performance)
      sceneView.setOnMouseMoved(null);
    };
```

File: src/main/java/com/esri/samples/featurelayers/change_feature_layer_renderer/ChangeFeatureLayerRendererSample.java [hunk 3]
```diff
@@ -20,12 +20,10 @@
 import javafx.geometry.Insets;
```
Comment (john0005) [inline src/main/java/com/esri/samples/featurelayers/change_feature_layer_renderer/ChangeFeatureLayerRendererSample.java:20]: Image in ReadME is now slightly different from sample.

File: src/main/java/com/esri/samples/featurelayers/feature_layer_definition_expression/FeatureLayerDefinitionExpressionSample.java [hunk 4]
```diff
@@ -20,12 +20,11 @@
 import javafx.geometry.Insets;
```
Comment (john0005) [inline src/main/java/com/esri/samples/featurelayers/feature_layer_definition_expression/FeatureLayerDefinitionExpressionSample.java:20]: Image in ReadME is now slightly different from sample.

**Review by john0005 — APPROVED** [review]
Looks good!

**Pull request merged**